### PR TITLE
Don’t compute onbuild triggers for images that are stage names

### LINKED ISSUE #502: Unhelpful error message
### Expected behavior

A helpful error message. 

### Actual behavior

I can't tell if it's coming from skaffold or docker cli:

```
$ skaffold dev -f foo/skaffold.yaml
WARN[0001] Error processing base image for onbuild triggers: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fpython%3Apull&service=registry.docker.io: exit status 1. Dependencies may be incomplete.
WARN[0001] Error processing base image for onbuild triggers: Get https://auth.docker.io/token?scope=repository%3Amicrosoft%2Fdotnet%3Apull&service=registry.docker.io: exit status 1. Dependencies may be incomplete.
WARN[0001] Error processing base image for onbuild triggers: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fnode%3Apull&service=registry.docker.io: exit status 1. Dependencies may be incomplete.
```

### Information

Skaffold version: v0.5.0
Operating system: macOS
Content of skaffold.yaml: 

```
apiVersion: skaffold/v1alpha2
kind: Config
build:
  artifacts:
  - imageName: vote
    workspace: voting-app/vote/
  - imageName: worker
    workspace: voting-app/worker/
  - imageName: result
    workspace: voting-app/result/
deploy:
  kubectl:
    manifests:
    - ./voting-app/k8s-specifications/*
```

### Steps to reproduce the behavior

  1. Start docker for mac
  2. Start minikube (xhyve driveRr)
  3. `~/.docker/config` contents:

```
{
  "credHelpers" : {
    "us.gcr.io" : "gcloud",
    "asia.gcr.io" : "gcloud",
    "gcr.io" : "gcloud",
    "staging-k8s.gcr.io" : "gcloud",
    "eu.gcr.io" : "gcloud"
  },
  "auths" : {
    "https://gcr.io" : {

    },
    "us.gcr.io" : {

    },
    "https://staging-k8s.gcr.io" : {

    },
    "k8s.gcr.io" : {

    },
    "https://eu.gcr.io" : {

    },
    "staging-k8s.gcr.io" : {

    },
    "gcr.io" : {

    },
    "asia.gcr.io" : {

    },
    "https://k8s.gcr.io" : {

    },
    "https://asia.gcr.io" : {

    },
    "https://us.gcr.io" : {

    },
    "eu.gcr.io" : {

    }
  },
  "HttpHeaders" : {
    "User-Agent" : "Docker-Client/18.03.0-ce (darwin)"
  },
  "experimental" : "enabled",
  "credsStore" : "osxkeychain",
  "orchestrator" : "swarm"
}
```
  4.  Run `skaffold dev` command above.
  5. deleting `~/.docker/config.json` made things proceed, but I get different warnings:

```
skaffold dev -f 10-local-build-run-with-minikube/skaffold.yaml
2018/05/01 23:31:46 Unable to read "/Users/ahmetb/.docker/config.json", falling back on anonymous: open /Users/ahmetb/.docker/config.json: no such file or directory
2018/05/01 23:31:47 manifest digest: "sha256:ee8b362b5afc1e9db00b79c4db7d7744986507a8ba1d88f53e531e46e93454ec" does not match Docker-Content-Digest: "sha256:a346d13efede199fe161e5146acf080a205256431d82cc26f845eeb9c47b7bf7" for "index.docker.io/library/python:2.7-alpine"
2018/05/01 23:31:48 Unable to read "/Users/ahmetb/.docker/config.json", falling back on anonymous: open /Users/ahmetb/.docker/config.json: no such file or directory
2018/05/01 23:31:49 manifest digest: "sha256:01528b7048c72d7a36f4c216ce1b2c60751308b27538c140f12e0b10ba392246" does not match Docker-Content-Digest: "sha256:f4ea9cdf980bb9512523a3fb88e30f2b83cce4b0cddd2972bc36685461081e2f" for "index.docker.io/microsoft/dotnet:2.0.0-sdk"
2018/05/01 23:31:49 Unable to read "/Users/ahmetb/.docker/config.json", falling back on anonymous: open /Users/ahmetb/.docker/config.json: no such file or directory
2018/05/01 23:31:50 manifest digest: "sha256:14b627a91c92566d489d9d9073e465563be0e0c598c9537aa32e871a812018f5" does not match Docker-Content-Digest: "sha256:6bb963d58da845cf66a22bc5a48bb8c686f91d30240f0798feb0d61a2832fc46" for "index.docker.io/library/node:8.9-alpine"
Starting build...
Found [minikube] context, using local docker daemon.
WARN[0004] run: build: build step: running build for artifact: running build: read auth configs: docker config: opening docker config: open /Users/ahmetb/.docker/config.json: no such file or directory
Watching for changes...
```

### PR DESCRIPTION
**Pull request by dgageot**
Fix #502

### CODE REVIEW

**Comment by dgageot** [comment]
ping @balopat @r2d4 Could you TAL?
In another PR, I'd like to refactor the code that parses the Dockerfiles

File: pkg/skaffold/docker/parse_test.go [hunk 1]
```diff
@@ -171,6 +176,10 @@ var ImageConfigs = map[string]*v1.ConfigFile{
 }
 
 func mockRetrieveImage(image string) (*v1.ConfigFile, error) {
+	if image == "base" {
+		panic("we should never retrieve images which are stage names")
```
Comment (balopat) [inline pkg/skaffold/docker/parse_test.go]: panic is considered a bad practice in testing as it stops the test execution - can we just return an `error`, or pass in `testing.T`?

Comment (balopat) [reply pkg/skaffold/docker/parse_test.go]: also assertion logic should not be embedded in the mock, this makes a test hard to understand.

Comment (balopat) [reply pkg/skaffold/docker/parse_test.go]: suggestion: reorganize the mock with usual "capture" semantics, that records the calls as a slice of strings and make the retrieved images an explicit assertion in the tests.

**Comment by dgageot** [comment]
@balopat should be all good now.

**Pull request merged**